x-pack/legacy/plugins/lens/server/usage/visualization_counts.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SavedObjectsClient, SavedObjectsFindResponse } from 'src/core/server';
+import { Document } from '../../public/persistence';
+import { LensUsage } from './types';
+
+function getTypeName(doc: Document): string {
+  if (doc.visualizationType === 'lnsXY') {
+    return (doc.state.visualization as { preferredSeriesType: string }).preferredSeriesType;
+  } else {
+    return doc.visualizationType as string;
+  }
+}
+
+function groupByType(response: SavedObjectsFindResponse) {
+  const byVisType: Record<string, number> = {};
+
+  response.saved_objects.forEach(({ attributes }) => {
```

Comment:
Nit: this feels like it should be a reduce, since that's basically what you're doing in the forEach.

```js
  return response.saved_objects.reduce((acc, { attributes }) => {
    const type = getTypeName(attributes as Document);
    byVisType[type] = (byVisType[type] || 0) + 1;
    return acc;
  }, {} as Record<string, number>);
```